IceCreamShop.playground/Contents.swift

```diff
@@ -1 +1,83 @@
+import UIKit
+
+struct Flavor {
+    var name: String
+    var rating: Double
+}
+
+enum Size: Double {
+    case small = 1.00
+    case medium = 2.88
+    case large = 5.25
+}
+
+struct Cone {
+    let flavor: Flavor
+    let topping: String?
+    let size: Size
+    var totalSales: Double = 0.0
+    
+    func eat() {
+        print("Yum, I love \(flavor.name)!)")
+    }
+}
+
+class IceCreamShop {
+    var flavors: [Flavor]
+    var sizes: [Size]
+    var toppings: [String?]
+
+    
+    init(flavors: [Flavor], sizes: [Size], toppings: [String?]) {
+        self.flavors = flavors
+        self.sizes = sizes
+        self.toppings = toppings
+ //       self.totalSales = totalSales
+    }
+}
+
+
+func listTopFlavors(_ flavors: [Flavor]) {
+    var flavors: [Flavor] = flavors
+    
+    for flavor in flavors {
+        if flavor.rating >= 4.0 {
+            flavors.append(contentsOf: flavors)
+            print("Our top flavors are \(flavor.name)!")
+        }
+    }
+}
+
+func orderCone(_ flavors: Flavor, toppings: String?, size: Size, totalSales: Double) -> Cone? {
```

Comment:
Just so you know, the underscore means that their will be no external name for the flavors parameter. So when you call it you will write:
``` swift
orderCone(yourFlavor, toppings: "some toppings", size: .large)
```

You could omit the underscore if you would like it to have the external name flavors, or you could put a different name in place of the _ such as `withFlavor`. It all comes down to preference.